src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,326 @@
+        :param driver_resources: If `embeddings_generation_driver_mode=BATCH` this can
+            be used to specify the driver resources.
+        :param driver_image: If `embeddings_generation_driver_mode=BATCH` this can be
+            used to specify the driver Docker image.
+        :param extra_driver_modules: If `embeddings_generation_driver_mode=BATCH` this
+            can be used to install extra pip package to the image.
+        :param max_tasks_per_stage: Number of maximum udf tasks per computation stage.
+        :param embeddings_generation_mode: TaskGraph execution mode for embeddings
+            generation.
+        :param embeddings_generation_driver_mode: TaskGraph execution mode for the
+            ingestion driver.
+        :param vector_indexing_mode: TaskGraph execution mode for the vector indexing.
+        :param index_update_kwargs: Extra arguments to pass to the index update job.
+        """
+        import tiledb
+
+        # Install extra packages not yet available in vectorsearch UDF image.
+        # TODO(nikos) remove the pacakge installations after the image is updated
+        def install_extra_worker_modules():
+            import os
+            import subprocess
+            import sys
+
+            sys.path.insert(0, "/home/udf/.local/bin")
+            sys.path.insert(0, "/home/udf/.local/lib/python3.9/site-packages")
+            os.environ["PATH"] = f"/home/udf/.local/bin:{os.environ['PATH']}"
+            subprocess.check_call(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "install",
+                    "--force-reinstall",
+                    "tiledb-vector-search==0.0.23",
+                    "tiledb==0.25.0",
+                    "tiledb-cloud==0.11.10",
+                ]
+            )
+            subprocess.check_call(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "install",
+                    "transformers==4.37.1",
+                    "PyMuPDF",
+                    "beautifulsoup4",
+                ]
+            )
+
+        install_extra_worker_modules()
+        import importlib
+
+        importlib.reload(tiledb)
+
+        from tiledb.vector_search.embeddings import SentenceTransformersEmbedding
+        from tiledb.vector_search.object_api import object_index
+        from tiledb.vector_search.object_readers import DirectoryTextReader
+
+        reader = DirectoryTextReader(
+            uri=file_dir_uri,
+            glob=f"**/{file_name}" if file_name is not None else glob,
+            exclude=exclude,
+            suffixes=suffixes,
+            text_splitter=text_splitter,
+            text_splitter_kwargs=text_splitter_kwargs,
+        )
+
+        embedding = SentenceTransformersEmbedding(
+            model_name_or_path=model_name_or_path,
+        )
+        if create_index:
```

Comment:
We should detect if the index_uri exists or not so the UX is cleaner for the user and the UI,

Reply:
Yes the UI will list our existing ones or let a user select a new one. That doesn't mean the code has to require the user to toggle it themselves. In other ingestors like VCF we have a simple check to see if the uri already exists or not to make it a better UX instead of having the user perform that check.